### PR TITLE
[specific ci=Group23-VIC-Machine-Service] Add DNS parsing in vch create API

### LINKED ISSUE #7017: VCH created by wizard with static public network fails to initialize
Found by @pdaigle 

Create a VCH using the wizard with static public network (instead of DHCP)
The VCH created fails to connect to docker endpoint, and admin page is not able to authenticate.

Create params:
```
{"name":"virtual-container-host","compute":{"cpu":{"limit":{"units":"MHz","value":0}},"memory":{"limit":{"units":"MiB","value":0}},"resource":{"name":"/RegionA01/host/RegionA01-COMP01"}},"storage":{"image_stores":["RegionA01-ISCSI01-COMP01"],"base_image_size":{"units":"GiB","value":8},"volume_stores":[{"datastore":"RegionA01-ISCSI01-COMP01","label":"default"}]},"network":{"bridge":{"ip_range":"172.16.0.0/12","port_group":{"name":"Bridge01-RegionA01-vDS-COMP"}},"public":{"port_group":{"name":"VM-RegionA01-vDS-COMP"},"static":"192.168.100.22/24","gateway":{"address":"192.168.100.1"},"nameservers":["192.168.110.10"]},"container":[{"port_group":{"name":"VM-RegionA01-vDS-COMP"},"alias":"routable","firewall":"open"}]},"auth":{"client":{"no_tls_verify":true},"server":{"generate":{"size":{"value":2048,"units":"bits"},"organization":["virtual-container-host"],"cname":"virtual-container-host"}}},"container":{},"debug":0,"endpoint":{"operations_credentials":{"user":"administrator@vsphere.local","password":"VMware1!"}},"registry":{}}
```

From the broken VCH, docker persona hangs waiting for portlayer to come up:
```
time="2018-01-02T19:00:03Z" level=info msg="Launching docker personality pprof server on 127.0.0.1:6062" 
Jan  2 2018 19:00:03.843Z ERROR Unable to load CAs for registry access in config
Jan  2 2018 19:00:03.849Z INFO  Waiting for portlayer to come up
```

Portlayer fails to connect to VC:
```
time="2018-01-02T19:00:03Z" level=info msg="Launching portlayer server pprof server on 127.0.0.1:6063" 
time="2018-01-02T19:00:03Z" level=info msg="{Formatter:0xc42011bde0 Level:info Syslog:<nil>}" 
Jan  2 2018 19:00:19.102Z FATAL configure_port_layer ERROR: Failed to connect to vcsa-01a.corp.local: Post https://vcsa-01a.corp.local/sdk: dial tcp: lookup vcsa-01a.corp.local: no such host
```

VCH log: [vch-logs-20180102.zip](https://github.com/vmware/vic/files/1598579/vch-logs-20180102.zip)

Cc: @zjs

### PR DESCRIPTION
**Pull request by AngieCris**
Fixes #7017 

Added DNS parsing in API create
DNS servers are supplied as `nameservers` field in each of the client, management and public networks. We only use the DNS servers for public network, and apply that for client and management config as well. 

@hickeng @mhagen-vmware What would be a good way to test API create with static public network and DNS servers? I have a nimbus VC but it doesn't allow me to reserve static IPs. The only way I can think of is to build an OVA and test it in OneCloud box via remote desktop

### CODE REVIEW

File: cmd/vic-machine/common/dns.go [hunk 1]
```diff
@@ -25,15 +25,15 @@ import (
 
 // general dns
 type DNS struct {
-	dns   cli.StringSlice `arg:"dns-server"`
+	Dns   cli.StringSlice `arg:"dns-server"`
```
Comment (jzt) [inline cmd/vic-machine/common/dns.go]: Should this be called `DNSServers` and use `arg:dns-servers`?

File: cmd/vic-machine/common/dns.go [hunk 2]
```diff
@@ -25,15 +25,15 @@ import (
...
 	IsSet bool
 }
 
 func (d *DNS) DNSFlags(hidden bool) []cli.Flag {
 	return []cli.Flag{
 		cli.StringSliceFlag{
 			Name:   "dns-server",
-			Value:  &d.dns,
+			Value:  &d.Dns,
 			Usage:  "DNS server for the client, public, and management networks. Defaults to 8.8.8.8 and 8.8.4.4 when VCH uses static IP",
```
Comment (jzt) [inline cmd/vic-machine/common/dns.go]: Should the usage specify how to enter multiple values (comma/space separated, etc)?

File: cmd/vic-machine/common/dns.go (hunk 1 above)
Comment (jzt) [reply cmd/vic-machine/common/dns.go]: Or maybe `NameServers`?

Comment (AngieCris) [reply cmd/vic-machine/common/dns.go]: just changed it to NameServers...go lint complained about the names

File: cmd/vic-machine/common/dns.go (hunk 2 above)
Comment (zjs) [reply cmd/vic-machine/common/dns.go]: To specify multiple servers I believe you specify the argument multiple times (`--dns 8.8.8.8 --dns 8.8.4.4`), but that's not clear from the current usage string.

Comment (AngieCris) [reply cmd/vic-machine/common/dns.go]: This is from CLI and for example if we have 3 DNS servers, we just give it 3 `dns-server` args:
```--dns-server=1.1.1.1 --dns-server=2.2.2.2 --dns-server=3.3.3.3```

**Comment by AngieCris** [comment]
@zjs Test added to simply check if DNS goes into VCH config. Functional test would require a real working DNS server tho

File: tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.robot [hunk 3]
```diff
@@ -164,6 +164,12 @@ Create complex VCH
     Output Should Contain    --container-network-dns=${PUBLIC_NETWORK}:8.8.4.4
     Output Should Contain    --container-network-firewall=${PUBLIC_NETWORK}:outbound
 
+    Output Should Contain    --public-network=${PUBLIC_NETWORK}
+    Output Should Contain    --public-network-gateway=192.168.100.1
+    Output Should Contain    --public-network-ip=192.168.100.22/24
+    Output Should Contain    --dns-server=192.168.110.10
```
Comment (zjs) [inline tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.robot]: Just for completeness, can you also make sure the DNS servers show up in the inspect API response bellow?

**Comment by mdubya66** [comment]
validated on @pdaigle's environment, per @AngieCris and @hickeng

**Comment by hickeng** [comment]
@mdubya66 If you merge PRs, please remember to strip the `[...]` tags from the commit title - our git history is bad enough without that kind of pollution added in.

**Pull request merged**